ucc/compile.py

```diff
@@ -47,14 +53,30 @@ def compile(
         return_format = get_program_type_alias(circuit)
 
     # Translate to Qiskit Circuit object
-    qiskit_circuit = transpile(circuit, "qiskit")
+    qiskit_circuit = translate(circuit, "qiskit")
     ucc_default1 = UCCDefault1(target_device=target_device)
     if custom_passes is not None:
         ucc_default1.pass_manager.append(custom_passes)
     compiled_circuit = ucc_default1.run(
         qiskit_circuit,
     )
 
+    if target_gateset is not None:
+        # Translate into user-defined gateset; no optimization
+        compiled_circuit = qiskit_transpile(
+            compiled_circuit, basis_gates=target_gateset, optimization_level=0
+        )
+    elif hasattr(target_device, "operation_names"):
+        # Use target_device gateset if available
+        target_gateset = target_device.operation_names
```

Comment:
I think we should raise an error or warning if a user passed `target_gateset` is different from the `target_device.operation_names`. IMO this could lead to unexpected behavior if a user is not careful. WDYT?